src/generate_pisa_index.cpp

```diff
@@ -21,19 +41,138 @@ std::ostream &write_sequence(std::ostream &os, gsl::span<T> sequence)
 }
 
 
+struct inverted_index {
+
+  std::unordered_map<uint32_t, std::vector<uint32_t>> m_documents {};
+  std::unordered_map<uint32_t, std::vector<uint32_t>> m_frequencies {};
+  std::map<uint32_t, uint32_t> m_document_sizes {};
```

Comment:
I understand you're using `std::map` because you want to access it in order at the end, but that's the only reason, right? If so, I would just stuff pairs into a vector, and then sort. It should be much more cache friendly

Reply:
Fair call! Will refactor this one.